packages/wallet-sdk/src/relay/MobileRelay.ts

```diff
@@ -1,41 +1,58 @@
-import { isInIFrame } from "../util";
+import { getLocation } from "../util";
 import { WalletLinkRelay, WalletLinkRelayOptions } from "./WalletLinkRelay";
 import { CancelablePromise } from "./WalletSDKRelayAbstract";
+import { Web3Method } from "./Web3Method";
+import { Web3Request } from "./Web3Request";
 import { RequestEthereumAccountsResponse } from "./Web3Response";
 
 export class MobileRelay extends WalletLinkRelay {
-  private _useMobileWalletLink: boolean;
+  private _enableMobileWalletLink: boolean;
 
   constructor(options: Readonly<WalletLinkRelayOptions>) {
     super(options);
-    this._useMobileWalletLink = options.useMobileWalletLink ?? false;
+    this._enableMobileWalletLink = options.enableMobileWalletLink ?? false;
   }
 
   // override
   public requestEthereumAccounts(): CancelablePromise<RequestEthereumAccountsResponse> {
-    if (this._useMobileWalletLink) {
+    if (this._enableMobileWalletLink) {
       return super.requestEthereumAccounts();
     }
 
     // TODO: Implement & present in-page wallet picker instead of navigating to www.coinbase.com/connect-dapp
     return {
       promise: new Promise(() => {
-        let location: Location;
-        try {
-          if (isInIFrame() && window.top) {
-            location = window.top.location;
-          } else {
-            location = window.location;
-          }
-        } catch (e) {
-          location = window.location;
-        }
-
+        const location = getLocation();
         location.href = `https://www.coinbase.com/connect-dapp?uri=${encodeURIComponent(
           location.href,
         )}`;
       }),
       cancel: () => {},
     };
   }
+
+  // override
+  protected publishWeb3RequestEvent(id: string, request: Web3Request): void {
+    super.publishWeb3RequestEvent(id, request);
+
+    if (this._enableMobileWalletLink) {
+      this.openCoinbaseWalletDeeplink(request);
+    }
+  }
```

Comment:
publishWeb3RequestEvent function has been overridden to open the cbwallet app when client sends web3 requests to walletlink. 

note that there might be a potential edge case in this code.
ideally, this logic should appended to publishEvent like below, being piped after publishEvent observable to get triggered when ws communication happens:
``` typescript
protected publishEvent(
    event: string,
    message: RelayMessage,
    callWebhook: boolean,
  ): Observable<string> {
    const observable = super.publishEvent(event, message, callWebhook);

    if (
      this._enableMobileWalletLink &&
      event === "Web3Request" &&
      message.type === RelayMessageType.WEB3_REQUEST
    ) {
      return observable.pipe(
        tap(() => {
          const web3RequestMessage = message as Web3RequestMessage;
          this.openCoinbaseWalletDeeplink(web3RequestMessage.request);
        }),
      );
    }

    return observable;
  }
```
however, we have encountered an issue where the browser seems to restrict calling window.open within the rxjs subscription block. 

if it can't be appended to the original observable, there's no point to modify it, since it's called with SESSION_ID_REQUEST and LINKED messages as well. (one of previous attempt below)
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/428200af-0f64-448f-9821-8c0dd3bfac8c) 

so to simplify the logic, overriding `publishWeb3RequestEvent` instead.